drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -142,7 +158,11 @@ static void spi_mcux_isr(const struct device *dev)
 	struct spi_mcux_data *data = dev->data;
 	SPI_Type *base = config->base;
 
+	pm_policy_device_power_lock_get(data->dev);
+
 	SPI_MasterTransferHandleIRQ(base, &data->handle);
+
+	pm_policy_device_power_lock_put(dev);
```

Comment:
in this whole file, you only need to call get when the SPI is starting a transfer, and call put when we are finishing it or erroring from the transfer, you don't need to guard every single function implementation with the get and put